shell/browser/native_window.cc

```diff
@@ -793,6 +844,226 @@ bool NativeWindow::IsTranslucent() const {
   return false;
 }
 
+void NativeWindow::DebouncedSaveWindowState() {
+  save_window_state_timer_.Start(
+      FROM_HERE, base::Milliseconds(200),
+      base::BindOnce(&NativeWindow::SaveWindowState, base::Unretained(this)));
+}
+
+void NativeWindow::SaveWindowState() {
+  if (!window_state_persistence_enabled_ || is_being_restored_)
+    return;
+
+  gfx::Rect bounds = GetBounds();
+  const display::Screen* screen = display::Screen::GetScreen();
+  DCHECK(screen);
+  // GetDisplayMatching returns a fake display with 1920x1080 resolution at
+  // (0,0) when no physical displays are attached.
+  // https://source.chromium.org/chromium/chromium/src/+/main:ui/display/display.cc;l=184;drc=e4f1aef5f3ec30a28950d766612cc2c04c822c71
+  const display::Display display = screen->GetDisplayMatching(bounds);
+
+  // Skip window state persistence when display has invalid dimensions (0x0) or
+  // is fake (ID 0xFF). Invalid displays could cause incorrect window bounds to
+  // be saved, leading to positioning issues during restoration.
+  // https://source.chromium.org/chromium/chromium/src/+/main:ui/display/types/display_constants.h;l=28;drc=e4f1aef5f3ec30a28950d766612cc2c04c822c71
+  if (display.id() == display::kDefaultDisplayId ||
+      display.size().width() == 0 || display.size().height() == 0) {
```

Comment:
This check doesn't seem to be firing correctly on Windows and Linux, at least in CI - happy to help look into why!